### PR TITLE
chore: Allows for specifying the docker tag of a component image as an environment variable

### PR DESCRIPTION
**Pull request by jchauncey**
This changes only the workflow-dev and workflow-dev-e2e charts for now.

### CODE REVIEW

File: workflow-dev/manifests/deis-logger-fluentd-daemon.yaml [hunk 1]
```diff
@@ -20,7 +21,7 @@ spec:
       serviceAccount: deis-logger-fluentd
       containers:
       - name: deis-logger-fluentd
-        image: quay.io/deisci/fluentd:canary
+        image: quay.io/deisci/fluentd:208d0dc
         imagePullPolicy: Always
         securityContext:
           privileged: true
```
Comment (arschles) [inline workflow-dev/manifests/deis-logger-fluentd-daemon.yaml:27]: can you remove this file and ensure that its templated version is in `workflow-dev/tpl`?

Comment (jchauncey) [inline workflow-dev/manifests/deis-logger-fluentd-daemon.yaml:27]: yup forgot to remove it. should be updated now

**Comment by jchauncey** [comment]
This PR also needs https://github.com/sgoings/chart-mate/pull/58

**Comment by jchauncey** [comment]
So heres the plan. Instead of us using the bump command in chart-mate which does a sed replace of the values in the chart. We will now use the environment variables specified in each template. 

The corresponding change in chart-mate sets each `COMPONENT_GIT_TAG` value to be `"git-${COMPONENT_SHA:0:7}"`

**Pull request merged**